src/halmos/bytevec.py

```diff
@@ -610,6 +611,23 @@ def set_word(self, offset: int, value: Word) -> None:
 
         self.set_slice(offset, offset + 32, value)
 
+    def concretize(self, substitution: dict[BitVecRef, BitVecRef]) -> None:
+        """
+        Replace all symbols in the chunks with their corresponding concrete values, if they exist in the given substituion mapping.
+        """
+        for offset, chunk in self.chunks.items():
+            chunk_data = chunk.data
+            if not isinstance(chunk, SymbolicChunk) or not is_expr_var(chunk_data):
```

Comment:
what is the purpose of `is_expr_var` here?

Reply:
it checks if it's simply a symbol, rather than an arbitrary symbolic term.

this is enough for the current calldata use cases. later, we may consider concretizing all symbols inside terms, but we'll need to make sure it doesn't hurt performance.

Reply:
cool, do you want to add it as a comment? I will definitely forget this :)

Reply:
added: https://github.com/a16z/halmos/pull/364/commits/9c46197d9263f24676ea0231961ec821b3f9f0a2